pkg/apis/datadoghq/v1alpha1/watermarkpodautoscaler_default.go

```diff
@@ -91,6 +91,9 @@ func CheckWPAValidity(wpa *WatermarkPodAutoscaler) error {
 	if wpa.Spec.Tolerance.MilliValue() > 1000 || wpa.Spec.Tolerance.MilliValue() < 0 {
 		return fmt.Errorf("tolerance should be set as a quantity between 0 and 1, currently set to : %v, which is %.0f%%", wpa.Spec.Tolerance.String(), float64(wpa.Spec.Tolerance.MilliValue())/10)
 	}
+	if wpa.Spec.ScaleUpLimitFactor == nil || wpa.Spec.ScaleDownLimitFactor == nil {
+		return fmt.Errorf("scaleuplimitfactor and scaledownlimitfactor can't be nil, make sure the WPA spec is defaulted")
```

Comment:
It might be worth rephrasing the second part about "defaulted", if a user sees this somewhere in the status it maybe misleading. Unless this error never surfaces that way?

Reply:
it should never been surfaced, because the WPA spec should be defaulted with the DefaultWPA method. This is more for the unit tests